### ISSUE TITLE
Excon cannot follow redirects with relative paths containing ".."

### ISSUE THREAD

**Issue by vaneyckt** [excon/excon#666]
Hey all,

given a webserver that has an endpoint of the form `<my_url>/foo/bar/` that redirects to `<my_url>/foo/bar` using a relative redirect containing "..", we notice that this redirect cannot be followed.

For example, imagine that `<my_url>/foo/bar/` responds with

```
< HTTP/1.1 301
< X-Frame-Options: SAMEORIGIN
< Location: ../bar
< Set-Cookie: wd-alt-sessionid=; Path=/; Secure; HttpOnly
< Content-Length: 0
< Date: Thu, 15 Mar 2018 09:55:03 GMT
< ...
```

then Excon will throw a 404 when trying to redirect.

```
require 'excon'

EXCON_MIDDLEWARES = Excon.defaults[:middlewares] + [Excon::Middleware::RedirectFollower]
response = Excon.get('http://localhost:4567/foo/bar/', ssl_verify_peer: false, middlewares: EXCON_MIDDLEWARES)
=> 404
```

Note that this redirect is successfully handled by both `curl` and Python's `request` library. Section 5.2.4 2A of the [RFC](https://tools.ietf.org/html/rfc3986#section-5.2.4) also states that this is valid.

**Comment by geemus** [comment]
I believe that ^ should fix things. Have another open bug I'd like to fix, but will try to follow up with a release soonish.

**Comment by vaneyckt** [comment]
thank you very much for the quick response on this! :rocket:

**Comment by geemus** [comment]
released as part of 0.61.0
